### DISCUSSION TITLE
useEffect not called in SSR components in production

### DISCUSSION THREAD

**Discussion by g-popovic** [vercel/next.js#29206] [Help]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Edge

### What operating system are you using?

Windows 10

### How are you deploying your application?

`next buid` => `node backend/server.js`

### Describe the Bug

I have a decently large Next.js/React application, connected to an Express.js backend server. A few days ago, some pages which worked flawlessly before, started behaving weirdly - more specifically, some `useEffect` hooks stopped firing completely. 

**More details:**

This is the file which is causing the bugs: 
```
export const AuthContext = createContext();

export function AuthProvider({ children }) {
  const [auth, setAuth] = useState(null);

  useEffect(() => {
    (async function () {
      const { data: result } = await axiosApp.get("/auth-status");
      setAuth(result);
    })();
  }, []);

  return !auth ? (
    <div className="pretend-this-is-a-progress-bar"></div>
  ) : (
    <AuthContext.Provider value={[auth, setAuth]}>{children}</AuthContext.Provider>
  );
}
```

As you can see I'm using react context here for the authentication in my app. This component here is located in the `_app.js` file, and the different next.js pages are passed as children of the `<AuthProvider/>`. 

As you can also see, this file sends a request to `/auth-staus` in it's `useEffect` hook, which tells the app whether the user is logged in or not. The problem is, **and I found this only occurs in pages that have a `getServerSideProps` function, and only in production - using a production build**, that this request is simply not sent - because the hook doesnt get called at all. What ends up happening is the `auth` state stays at `null` which in the end leads to the page loading infinitely for the user.

**Why** is this happening, and **what** can I do to solve this bug?

**PS**: Thank you in advance

### Expected Behavior

What I expect to happen (and what was happening a few days earlier) is that the `useEffect` hook gets called => axios request is sent => the response data is saved into the `auth` state => the app now knows whether the user is logged in.

### To Reproduce

Unfortunately the code base is private and I can't make the repo public. But as I said, it's a next.js app whose server is integrated with an express/node.js backend. When running a dev server (`nodemon backend/server.js`) this issue does not occur at all, but once I run `npm run build (next build)` and then `node backend/server.js` the issue begins. If any specific info is needed I will be here and ready to provide it, but like I mentioned I cannot share the entire repo.

**Comment by jpamarohorta** [comment]
I'm having a similar issue. In my case the useEffect (being used as an equivalent for a componentDidMount) doesn't trigger in **client side rendering**, only in **production** and my case only in **safari**. This behaviour only happens in certain pages, always after arriving to a page with a `router.push`. Not sure it's the same issue but it looks similar.
